app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,10 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.create(author_id: current_user.id, post_id: params[:post_id], text: params[:comment][:text])
+    if @comment.save
+      redirect_to user_post_path(params[:user_id], params[:post_id]), notice: 'Comment was successfully created.'
+    else
+      render :new
+    end
+  end
+end
```

Comment:
As per the project requirements, there should be a `new` method in the comments controller. I can see you have already added a route for the same in your routes.rb file. Also, in the views for comments, there should be a file `new.html.erb` instead of `_create.html.erb`. What you have done is created a partial view. I would recommend renaming your file to `new.html.erb`.